### PR TITLE
[CB-95] Implement examples limit

### PR DESCRIPTION
**Pull request by EPecherkin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +166,31 @@ TODO: Write good description for anyone who wants to customize behavior
...
+### Configuration
+
+#### Config file
+
+Create a YAML file for the runner. Default location is `./config/crystalball.yml` or simply `./crystalball.yml`. Or you can create it in another place but define `CRYSTALBALL_CONFIG` env variable with a path to it.
+You can find an example of a config file in [`spec/fixtures/crystalball.yml`](https://github.com/toptal/crystalball/blob/master/spec/fixtures/crystalball.yml)
+
+#### Environment variables
+
+Set `CRYSTALBALL_SKIP_MAP_CHECK` to `true` if you want to receive a warning about an outdated map
```
Comment (pluff) [inline README.md]: if you **_don't_** want

P.S. or maybe better `if you want to skip maps expiration period check`

File: README.md [hunk 2]
```diff
@@ -165,6 +166,31 @@ TODO: Write good description for anyone who wants to customize behavior
...
+Set `CRYSTALBALL_SKIP_EXAMPLES_LIMIT` to a desired amount of examples which can be run automatically. It is not recommended to leave this parameter unconfigured or to set it to an unreasonably high number.
```
Comment (pluff) [inline README.md]: This explanation is wrong. this param is also boolean and allows to skip examples limit check completely.

File: README.md [hunk 3]
```diff
@@ -165,6 +166,31 @@ TODO: Write good description for anyone who wants to customize behavior
...
+
+### Usage
+
+To build a prediction and run it just execute
+```ruby
+require 'crystalball'
+Crystalball::RSpec::Runner.invoke!
+```
+Or you can use copy and use [this helper](https://github.com/toptal/crystalball/blob/master/bin/crystalball)
```
Comment (pluff) [inline README.md]: Users don't have to copy or require anything manually. they can just do `bundle exec crystalball` and that's it

File: lib/crystalball/rspec/runner.rb [hunk 4]
```diff
@@ -64,6 +64,24 @@ def check_map(out)
...
       end
+
+      def run_specs(example_groups)
+        check_examples_limit(example_groups)
+        super
+      end
+
+      def check_examples_limit(example_groups)
+        limit = ENV['CRYSTALBALL_SKIP_EXAMPLES_LIMIT']&.to_i || config['examples_limit']
+        return unless limit&.positive?
```
Comment (pluff) [inline lib/crystalball/rspec/runner.rb]: incorrect logic.
```
limit = config['examples_limit'].to_i
return if ENV['CRYSTALBALL_SKIP_EXAMPLES_LIMIT'] || !limit.positive?
```

File: spec/fixtures/crystalball.yml [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # Custom path to your maps folder. Default: `tmp/execution_maps`
 map_path: 'execution_map.yml'
+# Maximum amount of examples which will be runned automaticly. Default: no limit.
```
Comment (pluff) [inline spec/fixtures/crystalball.yml]: **_automatically_**

**Review by pluff — CHANGES_REQUESTED** [review]

File: lib/crystalball/rspec/runner.rb (hunk 4 above)
Comment (jaimerson) [reply lib/crystalball/rspec/runner.rb]: I think naming is confusing here. If it's supposed to be an override for `config['examples_limit']`, it should be `ENV['CRYSTALBALL_EXAMPLES_LIMIT']`. If it should be a boolean to enable the limit or not, it should be `ENV['CRYSTALBALL_ENABLE_EXAMPLES_LIMIT']`. Pavel's review assumes it's the latter; for me it's fine either way, but take your pick because it's confusing right now.

File: spec/fixtures/crystalball.yml (hunk 5 above)
Comment (jaimerson) [reply spec/fixtures/crystalball.yml]: _**will run**_

**Review by jaimerson — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -165,6 +166,27 @@ TODO: Write good description for anyone who wants to customize behavior
...
+### Configuration
+
+#### Config file
+
+Create a YAML file for the runner. Default location is `./config/crystalball.yml` or simply `./crystalball.yml`. Or you can create it in another place but define `CRYSTALBALL_CONFIG` env variable with a path to it.
+You can find an example of a config file in [`spec/fixtures/crystalball.yml`](https://github.com/toptal/crystalball/blob/master/spec/fixtures/crystalball.yml)
+
+#### Environment variables
+
+`export CRYSTALBALL_CONFIG=path/to/crystalball.yml` if you want to override default path to cofig file.
```
Comment (jaimerson) [inline README.md]: `cofig` -> `config`

File: features/support/shared_contexts/simple_git_repository.rb [hunk 7]
```diff
@@ -36,7 +36,7 @@
   end
 
   def change(file_path)
-    file_path.write('changed')
+    file_path.write("'changed'")
```
Comment (jaimerson) [inline features/support/shared_contexts/simple_git_repository.rb:39]: Why was this necessary?

**Review by jaimerson — APPROVED** [review]

File: features/fixtures/simple_app/bin/crystalball [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'crystalball'
+require_relative '../standard_predictor'
+
+Crystalball::RSpec::Runner.invoke
```
Comment (pluff) [inline features/fixtures/simple_app/bin/crystalball]: I guess you don't need this file at all. You can specify proper predictor class in config

**Review by pluff — APPROVED** [review]
All good except of typo and `bin/crystalball` duplication

File: features/support/shared_contexts/simple_git_repository.rb (hunk 7 above)
Comment (EPecherkin) [reply features/support/shared_contexts/simple_git_repository.rb:39]: this file is loaded inside of `features/rspec_runner_spec.rb`
`changed` is not a valid ruby code :D
But `'changed'` is a valid string
